File: src/main/java/nl/vollo/kern/model/Medewerker.java
package nl.vollo.kern.model;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import javax.xml.bind.annotation.XmlRootElement;

@Entity
@Table(name = "medewerkers")
@XmlRootElement
@Getter
@Setter
@ToString
@EqualsAndHashCode(callSuper = true)
public class Medewerker extends DomainObject {

	private static final long serialVersionUID = 1L;

	@Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "vollo_seq")
    @SequenceGenerator(name = "vollo_seq", sequenceName = "vollo_seq", allocationSize = 1)
	@Column(name = "id", updatable = false, nullable = false)
	private Long id;

	@Version
	@Column(name = "version")
	private int version;

	@Column(nullable = false)
	private String voornaam;

	@Column
	private String tussenvoegsel;

	@Column(nullable = false)
	private String achternaam;

	public Medewerker() {
		super(DomainEntity.medewerker);
	}
}

File: src/main/java/nl/vollo/kern/model/Medewerker.kt
package nl.vollo.kern.model;

import javax.persistence.*
import javax.xml.bind.annotation.XmlRootElement

@Entity
@Table(name = "medewerkers")
@XmlRootElement
data class Medewerker(
        @Id
        @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "vollo_seq")
        @SequenceGenerator(name = "vollo_seq", sequenceName = "vollo_seq", allocationSize = 1)
        @Column(name = "id", updatable = false, nullable = false)
        override val id: Long? = null,

        @Version
        @Column(name = "version")
        override val version: Int = 0,

        @Column(nullable = false)
        val voornaam: String,

        @Column
        val tussenvoegsel: String? = null,

        @Column(nullable = false)
        val achternaam: String

) : DomainObject(DomainEntity.MEDEWERKER, id, version)
